code/service/src/main/scala/com/namely/chiefofstate/AggregateRoot.scala

```diff
@@ -178,40 +178,55 @@ object AggregateRoot {
       protosValidator: ProtosValidator,
       data: Map[String, com.google.protobuf.any.Any]): ReplyEffect[EventWrapper, StateWrapper] = {
 
-    val handlerOutput: Try[WriteHandlerHelpers.WriteTransitions] = commandHandler
-      .handleCommand(command, priorState)
-      .map(_.event match {
-        case Some(newEvent) =>
-          protosValidator.requireValidEvent(newEvent)
-          WriteHandlerHelpers.NewEvent(newEvent)
-
-        case None =>
-          WriteHandlerHelpers.NoOp
-      })
-      .flatMap({
-        case WriteHandlerHelpers.NewEvent(newEvent) =>
+    // state will be updated with the resulting state of each HandleEventResponse
+    var state = priorState.getState
```

Comment:
I would remove this line as it is an anti-pattern in scala.

Reply:
what would you suggest to do instead? what exactly is the anti pattern?

Reply:
You're mutating the "state" as you're running your code without a real reason to do so. In functional programming, you  typically want to pass around immutable objects.